### PR TITLE
[Image] zone resilient property addition

### PR DESCRIPTION
**Pull request by Lucretius**
Resolves https://github.com/terraform-providers/terraform-provider-azurerm/issues/1581

Updated documentation.  Note about where zone resiliency can be used comes from the [Azure documentation](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/capture-image-resource#create-an-image-of-a-vm-using-powershell)

Added a property to an existing test function which specifies the storage type so that it could be reused for the `zone_redundant` test.  Added a possible skip to the test, as not all zones support ZRS and did not want false positives interfering with the test.

### CODE REVIEW

**Comment by Lucretius** [comment]
This is not yet ready to merge.  I am having issues running the integration tests seeing the following:
```
azurerm_virtual_machine.testsource: Code="ImageNotFound" Message="The platform image 'Canonical:UbuntuServer:16.04-LTS:latest' is not available. Verify that all fields in the storage profile are correct."
```

Seems to be something local to my branch, I am working on it.

File: azurerm/resource_arm_image_test.go [hunk 1]
```diff
@@ -1388,3 +1433,14 @@ resource "azurerm_virtual_machine_scale_set" "testdestination" {
 }
 `, rInt, location, rInt, rInt, rInt, hostName, rInt, rInt, userName, password, rInt, userName, password, rInt)
 }
+
+// Region list pulled from https://docs.microsoft.com/en-us/azure/storage/common/storage-redundancy-zrs
+func supportsZRS(location string) bool {
+	zrsRegions := []string{"westus2", "francecentral", "southeastasia", "westeurope", "northeurope", "japaneast", "uksouth", "eastus", "eastus2", "centralus"}
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_image_test.go]: this only covers Azure Public (there's different Azure Regions which are supported e.g. Government, China, Germany) - as such I think we'd be best to remove this for the moment? (we're running the tests in some of these regions)

File: azurerm/data_source_image.go [hunk 2]
```diff
@@ -40,6 +40,11 @@ func dataSourceArmImage() *schema.Resource {
 
 			"location": locationForDataSourceSchema(),
 
+			"zone_resilient": {
```
Comment (tombuildsstuff) [inline azurerm/data_source_image.go]: could we set this field in the state here: https://github.com/terraform-providers/terraform-provider-azurerm/pull/3100/files#diff-21521ec4fe0d38353d87860312cd25a4R187

File: azurerm/resource_arm_image.go [hunk 3]
```diff
@@ -35,6 +35,13 @@ func resourceArmImage() *schema.Resource {
 
 			"resource_group_name": resourceGroupNameSchema(),
 
+			"zone_resilient": {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_image.go]: could we also Set this in the state here using the object returned from the API? https://github.com/terraform-providers/terraform-provider-azurerm/pull/3100/files#diff-95b4abb1c5040a8fd4f3c3674821b7b6R292

File: website/docs/r/image.html.markdown [hunk 4]
```diff
@@ -58,11 +58,14 @@ The following arguments are supported:
     the image. Changing this forces a new resource to be created.
 * `location` - (Required) Specified the supported Azure location where the resource exists.
     Changing this forces a new resource to be created.
+* `zone_resilient` - (Optional) Is zone resiliency enabled?  Defaults to `false`.  Changing this forces a new resource to be created.
```
Comment (tombuildsstuff) [inline website/docs/r/image.html.markdown]: we're trying to keep these sorted Required -> Optional and then Alphabetically within the fields; could we move this to below Tags (and above the `Note`?)

File: azurerm/resource_arm_image_test.go (hunk 1 above)
Comment (Lucretius) [reply azurerm/resource_arm_image_test.go]: Sure, my concern had been that I did not know in which locations these tests were going to be run, and did not want to have an acceptance test failure simply due to the fact that ZRS storage is not supported in some locations.

File: azurerm/data_source_image.go (hunk 2 above)
Comment (Lucretius) [reply azurerm/data_source_image.go]: Good catch thanks!

File: azurerm/resource_arm_image_test.go [hunk 5]
```diff
@@ -601,6 +641,7 @@ resource "azurerm_image" "test" {
   name                = "accteste"
   location            = "${azurerm_resource_group.test.location}"
   resource_group_name = "${azurerm_resource_group.test.name}"
+  zone_redundant      = %t
```
Comment (katbyte) [inline azurerm/resource_arm_image_test.go]: Should this be `zone_resilient`?
```suggestion
  zone_resilient      = %t
```

**Comment by Lucretius** [comment]
Ah nice catch @katbyte - I think I got mixed up looking at another resource which had the `zone_redundant` property

**Pull request merged**